packages/blade/src/components/Checkbox/_decisions/decisions.md

```diff
@@ -0,0 +1,220 @@
+| name          | `string`                    | No       | `undefined` | The name of the checkbox group, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name) |
+| isDisabled    | `boolean`                   | No       | `false`     | Control whether the checkbox group is disabled or not.                                                                             |
+| hasError     | `boolean`                   | No       | `false`     | Control whether the checkbox group is invalid or not.                                                                              |
+| showOptionalLabel    | `boolean`                   | No       | `false`     | Shows optional label in checkbox group header |
+
+### Examples:
+
+#### Basic
+
+```tsx
+<CheckboxGroup
+  label="Developers"
+  helpText="Pick blade developers"
+  defaultValue={['anurag', 'kamlesh']}
+>
+  <Checkbox value="anurag">Anurag</Checkbox>
+  <Checkbox value="kamlesh">Kamlesh</Checkbox>
+  <Checkbox value="chaitanya">Chaitanya</Checkbox>
+</CheckboxGroup>
+```
+
+#### Controlled, Uncontrolled
+
+```tsx
+const Controlled = () => {
+  const [selected, setSelected] = React.useState(['anurag', 'kamlesh']);
+
+  return (
+    <CheckboxGroup 
+      label="Developers (controlled)" 
+      value={selected} 
+      onChange={setSelected}
+    >
+      <Checkbox value="anurag">Anurag</Checkbox>
+      <Checkbox value="kamlesh">Kamlesh</Checkbox>
+      <Checkbox value="chaitanya">Chaitanya</Checkbox>
+    </CheckboxGroup>
+  );
+};
+
+const Uncontrolled = () => {
+  return (
+    <CheckboxGroup
+      label="Developers (uncontrolled)"
+      defaultValue={['anurag', 'chaitanya']}
+    >
+      <Checkbox value="anurag">Anurag</Checkbox>
+      <Checkbox value="kamlesh">Kamlesh</Checkbox>
+      <Checkbox value="chaitanya">Chaitanya</Checkbox>
+    </CheckboxGroup>
+  );
+};
+```
+
+### Note on CheckboxGroup
+
+For properties `isDisabled`, `hasError` we want the CheckboxGroup to pass down these props to all the Checkboxes inside of it.
+
+For showing optional label in `CheckboxGroup` we will introduce a prop `showOptionalLabel` which will only render a `(optional)` tag in the CheckboxGroupLabel,
+And the consumer will provide proper validation logic and depending on their use case and requirements they will pass `hasError` in conjunction with `showOptionalLabel` and they should also use `helpText` to convey the correct intent and extra information to the user.
+
+<img width="578" alt="Screenshot of checkbox with optional tag" src="https://user-images.githubusercontent.com/35374649/178486036-9371917f-b599-4f8b-b455-92d2c5f91c4b.png">
+
+
+**Example of user managed validation**
+
+Example of only one checkbox needs to be checked:
+
+```tsx
+function AnyOneHasToBeSelected() {
+  const [selected, setSelected] = React.useState([]);
+
+  return (
+    <CheckboxGroup
+      label="Do you love any of the fruites?"
```

Comment:
```suggestion
      label="Do you love any of the fruits?"
```

Reply:
lol it's always this spelling mistake 🙈 I should start using other examples like `Select your fav typescript feature`.